insights/collect.py

```diff
@@ -403,35 +403,38 @@ def generate_archive_name():
     return "insights-%s-%s" % (hostname, suffix)
 
 
-def collect(manifest=default_manifest, tmp_path=None, archive_name=None,
-            compress=False, rm_conf=None, client_config=None):
+def collect(client_config=None, rm_conf=None, tmp_path=None, archive_name=None,
+            compress=False):
     """
     This is the collection entry point. It accepts a manifest, a temporary
     directory in which to store output, and a boolean for optional compression.
 
     Args:
-        manifest (str or dict): json document or dictionary containing the
-            collection manifest. See default_manifest for an example.
+        client_config (InsightsConfig): Configurations read from insights-client
+            configuration, including "mainfest".
+        rm_conf (dict): Client-provided python dict containing keys
+            "commands", "files", and "keywords", to be injected
+            into the manifest blacklist.
         tmp_path (str): The temporary directory that is used to create a
             working directory for storing the final tar.gz if one is generated.
         archive_name (str): The directory that is used to generate the output
             as well as the final tar.gz.
         compress (boolean): True to create a tar.gz and remove the original
             workspace containing output. False to leave the workspace without
             creating a tar.gz
-        rm_conf (dict): Client-provided python dict containing keys
-            "commands", "files", and "keywords", to be injected
-            into the manifest blacklist.
-        client_config (InsightsConfig): Configurations read by the client tool.
 
     Returns:
         (str, dict): The full path to the created tar.gz or workspace.
         And a dictionary with relevant exceptions captured by the broker during
         core collection, this dictionary has the following structure:
         ``{ exception_type: [ (exception_obj, component), (exception_obj, component) ]}``.
     """
-
+    # Get the manifest from client configuration
+    manifest = default_manifest
+    if hasattr(client_config, 'manifest') and client_config.manifest:
+        manifest = client_config.manifest
```

Comment:
@xiangce   The manifest file load part from the origin code in [insights/client/core_collector.py  Line#35](https://github.com/RedHatInsights/insights-core/pull/4058/files#diff-76352cb529a81903277207d22b270de96a23532f285893c7f37e697c6f5109afL35) is skipped here. 

```
         if self.config.app is None:
                with open(self.config.manifest, 'r') as f:
                    manifest = f.read()
```

I see the `load_manifest(manifest)` calling below doesn't do the file load either. 

Then how do we load manifest from file?  Or we don't need the file load?

Reply:
The `load_mainfest` is strong enough to handle both `dict` and `file`:
https://github.com/RedHatInsights/insights-core/blob/729a1ca28e25468f60abd707bcaccb7b75bcb300/insights/collect.py#L280-L287

Reply:
Or, we should improve it to hand a `path` to a file.

Reply:
Yep, we still need to do the `open(mainfest, 'r')` for `load_mainfest`.